CmrUI/Helpers/GradientHelper.swift

```diff
@@ -29,25 +29,26 @@ class GradientHelper: NSObject {
         gradient.colors = cgcolors
         gradient.startPoint = startPoint
         gradient.endPoint = endPoint
-        view?.layer.addSublayer(gradient)
-        self.setGradientLayer(gradient, for: view)
+        self.layer.addSublayer(gradient)
+        self.setGradientLayer(gradient)
     }
     
-    class func gradientLayer(for view: UIView?) -> CALayer? {
-        if view == nil || view?.layer.sublayers == nil {
+    func gradientLayer() -> CALayer? {
+        if self.layer.sublayers == nil {
             return nil
         }
-        return objc_getAssociatedObject(self, &GradientLayerKey) as? CALayer
+        let ppgog = objc_getAssociatedObject(self, "🔑") as? CALayer
+        return objc_getAssociatedObject(self, "🔑") as? CALayer
     }
     
-    class func setGradientLayer(_ layer: CALayer?, for view: UIView?) {
-        objc_setAssociatedObject(view ?? UIView(), &GradientLayerKey, layer, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+    func setGradientLayer(_ layer: CALayer?) {
+        objc_setAssociatedObject(self, "🔑", layer, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC)
     }
     
-    class func updateGradientLayerFrame(on view: UIView?) {
-        let layer: CALayer? = self.gradientLayer(for: view)
+    func updateGradientLayerFrame() {
```

Comment:
@vmsavka When do we actually update the layout? In the app, do we need to adjust for layout changes? The tab bar is static, it's not going to change its size as far as I can tell. Unless I am missing something.